SU2_CFD/src/output/CFlowOutput.cpp

```diff
@@ -789,6 +789,530 @@ void CFlowOutput::Set_CpInverseDesign(CSolver *solver, const CGeometry *geometry
 
 }
 
+void CFlowOutput::Add_NearfieldInverseDesignOutput(){
+
+  AddHistoryOutput("EQUIVALENT_AREA",   "CEquiv_Area",  ScreenOutputFormat::SCIENTIFIC, "EQUIVALENT_AREA", "Equivalent area", HistoryFieldType::COEFFICIENT);
+  
+}
+
+void CFlowOutput::Set_NearfieldInverseDesign(CSolver *solver, const CGeometry *geometry, const CConfig *config){
+
+  bool output = true;
+  ofstream EquivArea_file, FuncGrad_file;
+  unsigned short iMarker = 0, iDim;
+  su2double Gamma, auxXCoord, auxYCoord, auxZCoord, InverseDesign = 0.0, DeltaX, Coord_i, Coord_j, jp1Coord, *Coord = nullptr, MeanFuntion,
+  *Face_Normal = nullptr, auxArea, auxPress, Mach, Beta, R_Plane, Pressure_Inf,
+  ModVelocity_Inf, Velocity_Inf[3], factor,
+  jFunction, jp1Function;
+  unsigned long jVertex, iVertex, iPoint, nVertex_NearField = 0, auxPoint,
+  auxDomain;
+  unsigned short iPhiAngle;
+  ofstream NearFieldEA_file; ifstream TargetEA_file;
```

Comment:
For all these variables please go through them and check whether it makes sense to declare them when they are used. E.g.
```c++
su2double ..., Mach, ... ;
Mach = config->GetMach();
```
to 
```c++
const su2double Mach = config->GetMach();
```
Also `const` them whenever possible (e.g. `nDim`). And don't worry about declaring a variable not at the top, that is totally ok (and even better imo) to keep the vars close to where they are needed

Please do that for all loop variables e.g. iDim, iPoint, iVertex unless you specificially need the var outside of its loop scope.
```c++
for(unsigned short iDim = 0; iDim < nDim; iDim++)
```

And feel free to use `auto` when you directly set a variable using an e.g. `Get<Value>` function e.g.
```c++
const auto nDim = geometry->GetnDim();
```
But do not use `auto` for anything that would be an `su2double`! (I know that might sound strange but there is an explanation to which I could give upon request)

Reply:
I think we could be a little bit less demanding with a first-time contributor that is already cutting a chunk out of the output_legacy :smile:

Reply:
Yeah, I think it is ok to not do all that. On the other hand, this is the easier and rewarding cleanup part imo :D (especially compared to non blocking MPI comms... Personally I would prefer to procrastinate with cleaning variables every day of the week 🤫 )

Reply:
The issue related to non blocking MPI comms either causes an error or stops execution unless I use a certain number of cores, so it has a higher priority for me. I'll address the rest of comments once I resolve such roadblocks.

Reply:
I have changed declaration for easy ones. I'm not sure which ones make sense to be declared when they are used. For iDim, iPoint, and iVertex, they are used in multiple "for" loops. I'm not sure if they should be declared in multiple "for" loops or declared only once at the top.

Reply:
Only for completeness and future developments (not something you have to address now IMO), these are the guidelines we should try to follow regarding variables declaration and initialization: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-introduce

Reply:
Thank you for the reference. I will work on it in the future.